sound/soc/codecs/pcm512x.c

```
@@ -1503,6 +1506,121 @@ const struct regmap_config pcm512x_regmap = {
 };
 EXPORT_SYMBOL_GPL(pcm512x_regmap);
 
+static int pcm512x_gpio_get_direction(struct gpio_chip *chip,
+				      unsigned int offset)
+{
+	struct pcm512x_priv *pcm512x = gpiochip_get_data(chip);
+	unsigned int val;
+	int ret;
+
+	ret = regmap_read(pcm512x->regmap, PCM512x_GPIO_EN, &val);
+	if (ret < 0)
+		return ret;
+
+	val = (val >> offset) & 1;
+
+	/* val is 0 for input, 1 for output, return inverted */
+	return !val;
+}
+
+static int pcm512x_gpio_direction_input(struct gpio_chip *chip,
+					unsigned int offset)
+{
+	struct pcm512x_priv *pcm512x = gpiochip_get_data(chip);
+	unsigned int mask;
+
+	mask = 1 << offset;
+
+	regmap_update_bits(pcm512x->regmap, PCM512x_GPIO_EN,
+			   mask, 0);
```

I think you should propagate the possible error here with `return regmap_update_bits(...);`? Also optionally you could just use `BIT(offset)` here directly as an argument and remove `mask`

---

yes, will do